core/src/main/java/com/linecorp/armeria/server/http/dynamic/Methods.java

```diff
@@ -0,0 +1,217 @@
+     *
+     * @see Options
+     * @see Get
+     * @see Head
+     * @see Post
+     * @see Put
+     * @see Patch
+     * @see Delete
+     * @see Trace
+     */
+    private static EnumSet<HttpMethod> httpMethods(Method method) {
+        EnumSet<HttpMethod> ret = EnumSet.noneOf(HttpMethod.class);
+        for (Annotation annotation : method.getAnnotations()) {
+            HttpMethod mappedMethod = HTTP_METHOD_MAP.get(annotation.annotationType());
+            if (mappedMethod != null) {
+                ret.add(mappedMethod);
+            }
+        }
+
+        return ret;
+    }
+
+    /**
+     * Returns the {@link DynamicPath} instance mapped to {@code method}.
+     */
+    private static DynamicPath dynamicPath(Method method) {
+        Path mapping = method.getAnnotation(Path.class);
+        String mappedTo = mapping.value();
+        return DynamicPath.of(mappedTo);
+    }
+
+    /**
+     * Returns the {@link ResponseConverter} instance from {@link Converter} annotation of the given
+     * {@code method}. The {@link Converter} annotation marked on a method can't be repeated and should not
+     * specify the target class.
+     */
+    private static ResponseConverter converter(Method method) {
+        Converter[] converters = method.getAnnotationsByType(Converter.class);
+        if (converters.length == 0) {
+            return null;
+        }
+        if (converters.length == 1) {
+            Converter converter = converters[0];
+            if (converter.target() != Object.class) {
+                throw new IllegalArgumentException(
+                        "@Converter annotation can't be marked on a method with a target specified.");
+            }
+            ResponseConverter ret = null;
+            try {
+                ret = converter.value().newInstance();
+            } catch (Exception e) {
+                throw new IllegalStateException(e);
+            } finally {
+                return ret;
+            }
+        }
+
+        throw new IllegalArgumentException("@Converter annotation can't be repeated on a method.");
+    }
+
+    /**
+     * Returns a mapping from {@link Class} to {@link ResponseConverter} instances from {@link Converter}
+     * annotations of the given {@code clazz}. The {@link Converter} annotation marked on {@code clazz} must
+     * specify the target class, except {@link Object}.class.
+     */
+    private static Map<Class<?>, ResponseConverter> converters(Class<?> clazz) {
+        Converter[] converters = clazz.getAnnotationsByType(Converter.class);
+        ImmutableMap.Builder<Class<?>, ResponseConverter> builder = ImmutableMap.builder();
+        for (Converter converter : converters) {
+            Class<?> target = converter.target();
+            if (target == Object.class) {
+                throw new IllegalArgumentException(
+                        "@Converter annotation can't be marked on a class without a target specified.");
```

Comment:
- can't be marked on a class without a target specified -> must have a target type specified